lualib/rspamadm/clickhouse.lua

```diff
@@ -0,0 +1,247 @@
+	if is_spam then
+          known_symbols[sym].seen_spam = known_symbols[sym].seen_spam + 1
+        else
+          known_symbols[sym].seen_ham = known_symbols[sym].seen_ham + 1
+        end
+      end
+    end
+  end
+
+  local known_symbols_list = {}
+  for k, v in pairs(known_symbols) do
+    local lower_count, higher_count
+    if v.seen_spam > v.seen_ham then
+      lower_count = v.seen_ham
+      higher_count = v.seen_spam
+    else
+      lower_count = v.seen_spam
+      higher_count = v.seen_ham
+    end
+    if lower_count / higher_count >= 0.95 then
+      -- Weak ham/spam correlation, remove symbol
+      print(string.format("Removing %s - weak ham/spam correlation", k))
+      known_symbols[k] = nil
+      removed = removed + 1
+    elseif v.seen / seen_total >= 0.9 then
+      -- Symbol is omnipresent, remove it
+      print(string.format("Removing %s - omnipresent symbol", k))
+      known_symbols[k] = nil
+      removed = removed + 1
+    else
+      known_symbols_list[v.id] = {
+        seen = v.seen,
+        name = k,
+      }
+    end
+  end
+
+  -- Walk results again & count correlations
+  local correlations = {}
+  for _, r in ipairs(res) do
+    for _, sym in ipairs(r['Symbols.Names']) do
+      for _, inner_sym_name in ipairs(r['Symbols.Names']) do
+        if inner_sym_name ~= sym then
+          local known_sym = known_symbols[sym]
+          local inner_sym = known_symbols[inner_sym_name]
+          if known_sym and inner_sym then
+            if not correlations[known_sym.id] then
+              correlations[known_sym.id] = {}
+            end
+            local n = correlations[known_sym.id][inner_sym.id] or 0
+            n = n + 1
+            correlations[known_sym.id][inner_sym.id] = n
+          end
+	end
+      end
+    end
+  end
+
+  -- Walk correlation matrix and divide by total counts
+  local remove = {}
+  for sym_id, row in pairs(correlations) do
+    for inner_sym_id, count in pairs(row) do
+      local known = known_symbols_list[sym_id]
+      if known and count == known.seen then
+        print(string.format("Removing %s - overlapped with %s",
+            known.name, known_symbols_list[inner_sym_id].name))
+        remove[known.name] = true
+      end
+    end
+  end
+
+  for k in pairs(remove) do
+    known_symbols[k] = nil
+    removed = removed + 1
+  end
+
+  print(string.format("Filtered %d of %d known symbols",
+      removed, symbols_count - 1))
+  for k in pairs(known_symbols) do print(k) end
+end
+
+local command_handlers = {
+  neural_profile = handle_neural_profile,
+}
+
+local function handler(args)
+  local cmd_opts = parser:parse(args)
+
+  load_config(cmd_opts.config_file)
+  local cfg_opts = rspamd_config:get_all_opt('clickhouse')
+
+  local function override_settings(params)
+    for _, which in ipairs(params) do
+      if cmd_opts[which] == nil then
+        cmd_opts[which] = cfg_opts[which]
+      end
+    end
+  end
+
+  override_settings({
```

Comment:
why not lua_util.override_defaults?

Reply:
Probably I forgot about it but reason seems to be that these fields don't promise to be set anywhere.